src/main/scala/edu/cmu/cs/obsidian/typecheck/Checker.scala

```diff
@@ -1789,7 +1790,7 @@ class Checker(globalTable: SymbolTable, verbose: Boolean = false) {
                         val fieldAST = newStateTable.lookupField(f)
                         if (fieldAST.isDefined) {
                             val (t, contextPrime2, ePrime) = inferAndCheckExpr(decl, contextPrime, e, consumptionModeForType(fieldAST.get.typ))
-                            newUpdates = newUpdates :+ (ReferenceIdentifier(f, Some(t)), ePrime)
+                            newUpdates = newUpdates :+ (ReferenceIdentifier(f, Some(t)), ePrime) // todo iev: does ePrime here need its obstype field updated?
```

Comment:
If inferAndCheckExpr is supposed to do it, then no. You should decide and update it if needed.